inc/az_ulib_ipc_api.h

```diff
@@ -185,25 +244,60 @@ AZ_NODISCARD az_result az_ulib_ipc_unpublish(
  * @brief   Try get an interface handle by the name from the IPC.
  *
  * This API tries to find an interface that fits the provided name. If there is an interface that
- * fits this criteria, this API will return its handle. If no published procedure fits it, this API
- * will return #AZ_ERROR_ITEM_NOT_FOUND.
+ * fits this criteria, this API will return its handle.
  *
  * Get a interface handle will increment the number of references to this interface, which means
  * that the IPC will know how many components is using this interface. When a component does not
- * need this interface anymore, it shall release it by calling az_ulib_ipc_release_interface().
+ * need this interface anymore, it shall release the handle by calling
+ * az_ulib_ipc_release_interface().
  *
  * @note    **Do not release an interface will cause memory leak.**
  *
- * @param[in]   name              The `az_span` with the interface name.
- * @param[in]   version           The #az_ulib_version with the desired version.
- * @param[in]   match_criteria    The #az_ulib_version_match_criteria with the match criteria for
- *                                the interface version.
+ * An interface is identified by 3 characteristics.
+ *  1. device:      Identifies **where** the interface is on a complex device that is composed by
+ *                  multiple CPUs.
+ *  2. package:     Identifies the **implementation** of the capabilities.
+ *  3. interface:   Identifies the **signature** of the capabilities.
+ *
+ * This function returns the handle of the interface that best fits the requirements following the
+ * follow rules. If the criteria doesn't match, this function will return #AZ_ERROR_ITEM_NOT_FOUND.
+ *
+ *  - interface_name is mandatory.
+ *  - interface_version
+ *      - If provided, this function will look up only for the interface with the interface_version.
+ *      - If #AZ_ULIB_VERSION_ANY (0), this function will look up for the interface with the lowest
+ *          version, which matches the other criteria.
```

Comment:
After play a little with it I realized that it is too dangerous to keep interface_version flexible. I'll make it mandatory as well.